File: python/examples/06_usage.py
"""Example 06 — Discovery + usage (Python)."""

from __future__ import annotations

import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from lawdiver import LawDiverClient

with LawDiverClient() as client:
    print("=== Discovery (public) ===")
    discovery = client.discovery()
    print(json.dumps(discovery, indent=2)[:800] + "\n…\n")

    print("=== Usage (authenticated) ===")
    usage = client.usage(30)

consumer = usage.get("consumer") or {}
pricing = usage.get("yourPricing") or {}
print(f"consumer: {consumer.get('name')} <{consumer.get('email')}> [{consumer.get('status')}]")
print(f"periodDays: {usage.get('periodDays')} since {usage.get('since')}")
print(f"rateLimitPerMinute: {pricing.get('rateLimitPerMinute')}")
print(f"maxCasesPerSearch: {pricing.get('maxCasesPerSearch')}")
print()
print("byOperation:")
for row in usage.get("byOperation") or []:
    print(
        f"  {row['operation']}: calls={row['calls']} units={row['units']} costCents={row['costCents']}"
    )
print(f"totalCostCents: {usage.get('totalCostCents')}")
print(f"requestId: {usage.get('requestId')}")

File: typescript/examples/06-usage.ts
/**
 * Example 06 — Discovery + usage ledger
 * Language: TypeScript (Node.js)
 *
 * Fastest way to separate network problems from bad credentials:
 * discovery (no key) vs usage (requires key).
 */
import { LawDiverClient } from "../src/client.js";

const client = new LawDiverClient();

console.log("=== Discovery (public) ===");
const discovery = await client.discovery();
console.log(JSON.stringify(discovery, null, 2).slice(0, 800) + "\n…\n");

console.log("=== Usage (authenticated) ===");
const usage = await client.usage(30);
console.log(`consumer: ${usage.consumer.name ?? "?"} <${usage.consumer.email ?? "?"}> [${usage.consumer.status}]`);
console.log(`periodDays: ${usage.periodDays} since ${usage.since}`);
console.log(`rateLimitPerMinute: ${usage.yourPricing.rateLimitPerMinute}`);
console.log(`maxCasesPerSearch: ${usage.yourPricing.maxCasesPerSearch}`);
console.log("");
console.log("byOperation:");
for (const row of usage.byOperation) {
  console.log(`  ${row.operation}: calls=${row.calls} units=${row.units} costCents=${row.costCents}`);
}
console.log(`totalCostCents: ${usage.totalCostCents}`);
console.log(`requestId: ${usage.requestId}`);
